src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPageInternalBase.vb

```diff
@@ -135,14 +135,18 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                     Dim supportedFrameworks As IEnumerable(Of TargetFrameworkMoniker) = TargetFrameworkMoniker.GetSupportedTargetFrameworkMonikers(vsFrameworkMultiTargeting, DTEProject, supportedTargetFrameworksDescriptor?.Converter)
 
                     'If the list doesn't contain any tfm, it means the project can't retarget.
-                    If Not supportedFrameworks.Any() Then
+
+                    Dim hasSupportedFramework = False
+
+                    For Each supportedFramework As TargetFrameworkMoniker In supportedFrameworks
+                        targetFrameworkComboBox.Items.Add(supportedFramework)
+                        hasSupportedFramework = True
+                    Next
+
+                    If Not hasSupportedFramework Then
                         targetFrameworkSupported = False
                         targetFrameworkComboBox.Items.Clear()
```

Comment:
It looks like this can never be useful? `targetFrameworkComboBox.Items.Clear()` was called at the beginning of this function, and no items should have been added in between.

Reply:
I think you're right, but I'm really reluctant to change this code as we don't have any tests for it and it's used infrequently. Given your other suggestion, I'm happy enough with how this code landed for now.